### PR TITLE
sd-pull-request

### PR DESCRIPTION
**Pull request by setholito**

### CODE REVIEW

File: src/01_nearest_larger.js [hunk 1]
```diff
@@ -1,5 +1,43 @@
...
+    }
 
+    // If no conditions are met, increment by 1 to continue search
+    distance += 1;
+  }
+  // If no conditions are met, and there's no larger number, return null
+  return null;
 }
 
 module.exports = nearestLarger;
```
Comment (canaandavis) [inline src/01_nearest_larger.js]: Nice updates here.  Does this make more sense now that you've seen the solution and worked through it?

Comment (setholito) [inline src/01_nearest_larger.js]: Yes! Seeing the solution was very helpful.

File: src/02_no_repeats.js [hunk 2]
```diff
@@ -1,5 +1,70 @@
+/**
+ * Takes two years and return the years that don't have repeat digits
+ * @function
+ * @param {integer} Beginning year
+ * @param {integer} End year
+ * @return {array} Array of years without duplicate digits
+ */
+
+// Integer, Integer >> Array
 function noRepeats(yearStart, yearEnd) {
```
Comment (canaandavis) [inline src/02_no_repeats.js]: Nice job here Seth.  Though, this solution does feel a bit verbose, have you had a chance to review the posted solution yet.  If so, how could you take some of the concepts used there to make your solution a bit more concise?

I have to give you props on your comments.  Even though there is a lot going on here your comments let me know that you understand it, keep up the great work!

File: src/02_no_repeats.js [hunk 3]
```diff
@@ -1,5 +1,70 @@
...
+ * Takes in a number, loops over it, and determines if there are duplicate digits or not
+ * @function
+ * @param {number} Integer, a year.
+ * @return {Boolean}
+ */
 
+// Integer >> Boolean
+function inspectNum(number){
+  // Make integer a string, use split to turn it into an array
+  var numberArr = String(number).split("");
```
Comment (canaandavis) [inline src/02_no_repeats.js]: In Javascript prefer `toString` over the constructor `String(n)` they may appear to do the same thing, but they work a bit different under the hood.

File: src/03_letter_count.js [hunk 4]
```diff
@@ -1,5 +1,31 @@
...
+  var letterCount = {};
+
+  // Split string into an array
+  var strArr = str.split('');
+
+  // Loop over the strArr
+  strArr.forEach(function(letter){
+    // If letter is not in the object, add it to letterCount object
+    // and give it a value of 1
+    if(letterCount[letter] === undefined){
```
Comment (canaandavis) [inline src/03_letter_count.js]: This solution looks great, I'll give one critique though.
`if (letterCount[letter] === undefined)` It isn't necessary to explicitly check for undefined here, the reason being is that if our letter is in the array we know it has a `truthy` value, which we've set to `1`.  So you could simply evaluate `if (!letterCount[letter])` or if you do not like checking on the negative and would prefer checking on the affirmative.

```
if(letterCount[letter]){
  letterCount[letter] += 1;
} else {
  if (letter !== ' ') {
    letterCount[letter] = 1;
   }  
}
```

But this is just me nitpicking, once again comments are great, keep up the great work!

File: src/04_ordered_vowels.js [hunk 5]
```diff
@@ -1,5 +1,41 @@
+
+// String >> String
```
Comment (canaandavis) [inline src/04_ordered_vowels.js]: Thanks for the updates here Seth.

Great work it looks like you have a pretty good understanding of what is going on here.  I know you didn't get this in until after the solution was posted, but it seems like you have a pretty good idea of the flow that is happening here.

Keep up the great work!

File: src/05_wonky_coins.js [hunk 6]
```diff
@@ -1,7 +1,17 @@
 var flatten = require('array-flatten');
 
-function wonkyCoins(n) {
-
+// Number >> Array
+function wonkyCoins(n){
```
Comment (canaandavis) [inline src/05_wonky_coins.js]: Looks good here, you understand what is happening?

File: src/06_morse_code.js [hunk 7]
```diff
@@ -1,5 +1,66 @@
+// morseEncode takes a sentence, splits it into an array
+// delimits using spaces, a forEach loops over each word
+// morseEncodeWord() turns each word into morse code and appends to sentence array
+// once the loop has completed we join everything back together with spaces.
 function morseEncode(n) {
```
Comment (canaandavis) [inline src/06_morse_code.js]: This looks good great job!

**Pull request closed without merging**